### PR TITLE
[v2] Preserve channels_last memory format in pad (#9560)

### LINKED ISSUE #9560: `torchvision.transforms.v2.functional.pad` should have maintained the `channels_last` format of the input
### 🐛 Describe the bug

```py
import torch
import torchvision.transforms.v2.functional
from torchcodec.decoders import VideoDecoder

a=VideoDecoder(r"F:\....gif",device='cuda')

# When input is CHW channels_last 
c=torchvision.transforms.v2.functional.crop(a[0], 10,10,200,282)
c.shape, c.stride()
# (torch.Size([3, 200, 200]), (1, 876, 3))

d=torchvision.transforms.v2.functional.resize(a[0], (300,300))
d.shape, d.stride()
# (torch.Size([3, 300, 300]), (1, 900, 3))

e=torchvision.transforms.v2.functional.pad(a[0], (500,500), 255, 'constant')
e.shape, e.stride()
# (torch.Size([3, 1241, 1292]), (1603372, 1292, 1))  # channels_last broken!

f=torch.nn.functional.pad(a[0], (500,500,500,500), 'constant', 255)
f.shape, f.stride()
# (torch.Size([3, 1241, 1292]), (1603372, 1292, 1))  # channels_last broken!

g=torchvision.transforms.v2.functional.crop(a[0], 10,10,500,500)
g.shape, g.stride()
# (torch.Size([3, 500, 500]), (250000, 500, 1))


# When input is NCHW channels_last 
aa=a[0].unsqueeze(0)
aa.shape,aa.stride()

cc=torchvision.transforms.v2.functional.crop(aa, 10,10,200,200)
print(cc.shape, cc.stride())

dd=torchvision.transforms.v2.functional.resize(aa, (300,300))
print(dd.shape, dd.stride())

ee=torchvision.transforms.v2.functional.pad(aa, (500,500), 255, 'constant')
print(ee.shape, ee.stride())

ff=torch.nn.functional.pad(aa, (500,500,500,500), 'constant', 255)
print(ff.shape, ff.stride())

gg=torchvision.transforms.v2.functional.crop(aa, 10,10,500,500)
print(gg.shape, gg.stride())

# torch.Size([1, 3, 200, 200]) (3, 1, 876, 3)
# torch.Size([1, 3, 300, 300]) (3, 1, 900, 3)
# torch.Size([1, 3, 1241, 1292]) (4810116, 1603372, 1292, 1)  # channels_last broken!
# torch.Size([1, 3, 1241, 1292]) (4810116, 1603372, 1292, 1)  # channels_last broken!
# torch.Size([1, 3, 500, 500]) (750000, 250000, 500, 1)


# When input is NCHW channels_last for sure! perfect and stictly channels_last stride.
aa=a[0].unsqueeze(0).to(memory_format=torch.channels_last)
aa.shape,aa.stride()

cc=torchvision.transforms.v2.functional.crop(aa, 10,10,200,200)
print(cc.shape, cc.stride())

dd=torchvision.transforms.v2.functional.resize(aa, (300,300))
print(dd.shape, dd.stride())

ee=torchvision.transforms.v2.functional.pad(aa, (500,500), 255, 'constant')
print(ee.shape, ee.stride())

ff=torch.nn.functional.pad(aa, (500,500,500,500), 'constant', 255)
print(ff.shape, ff.stride())

gg=torchvision.transforms.v2.functional.crop(aa, 10,10,500,500)
print(gg.shape, gg.stride())

# torch.Size([1, 3, 200, 200]) (211116, 1, 876, 3)
# torch.Size([1, 3, 300, 300]) (3, 1, 900, 3)
# torch.Size([1, 3, 1241, 1292]) (4810116, 1603372, 1292, 1)  # channels_last broken!
# torch.Size([1, 3, 1241, 1292]) (4810116, 1, 3876, 3)
# torch.Size([1, 3, 500, 500]) (750000, 250000, 500, 1)
```

As we can see, 
* `resize` and `crop` they both support "CHW" and "NCHW", "channels_last", "strict stride channels_last" memory_format.
* `torch.nn.functional.pad` supports only "NCHW", "strict stride channels_last" memory_format, not "CHW" nor "(unstrict) channels_last".
* `torchvision.transforms.v2.functional.pad` does not even support what `torch.nn.functional.pad` supported.

The "channels_last" is needed and has higher performance in image process where R G B are together so filters can process them at onece, and gpu need it to accelarate in its tensor cores.

So `torchvision.transforms.v2.functional.pad` should have the ablity to keep the `channels_last` format of the input like `resize` and `crop`, instead of breaking it within the pipeline process in torchvision.

Thank your time for reading.
I would like to make a PR.

### Versions

torchvision.__version__
'0.27.1+cu126'

### PR DESCRIPTION
**Pull request by FinalSunFlower**
Fixes #9560

### Summary
- Preserve an unambiguous channels-last layout through v2 image padding, including scalar and vector fills.
- Restore channels-last before reshaping back to the original image dimensions.
- Add CPU regression coverage for constant, edge, reflect, symmetric, and vector-fill padding through both F.pad and F.pad_image.

### Tests
- TestPad: 202 passed, 128 skipped, 18 xfailed.
- New channels-last regression: 10 passed.
- git diff --check: passed.

Full-suite execution is affected by local environment limitations (missing JPEG extension support, pytest-mock, and TorchScript source inspection under the command wrapper).

### CODE REVIEW

**Review by fallenmi — APPROVED** [review]
Reviewed exact head `3e4131cf53a34968123775f519fc0fd209102d80` against exact base `ac8d215f7d45d6601451b62e9f81622dac8aa0b4`.

The preservation point looks correct: the code detects an unambiguous channels-last layout after collapsing arbitrary leading dimensions to the kernel's 4D view, then restores that layout after the selected padding implementation. The vector-fill path retains it while applying the per-channel border values.

I exercised 228 CPU cases across 3D, 4D, and 5D image shapes; float32 and uint8; contiguous, strict channels-last, and row-padded channels-last-like inputs; scalar and vector fills; constant, edge, reflect, and symmetric modes; and positive and negative padding. Exact base, head, and the current GitHub merge produced byte-identical values in every case. All 66 unambiguous channels-last cases retain channels-last output on head and merge, while base loses it in 32. Scalar- and vector-fill autograd outputs and gradients also match base byte-for-byte.

The complete `TestPad` slice passes on all three revisions: base 192 passed / 128 skipped / 18 xfailed; head and merge 202 passed / 128 skipped / 18 xfailed. `git diff --check` is clean, and the current merge tree is byte-identical to the reviewed head.

GitHub currently exposes only the green CLA/import/internal-only checks rather than the normal public test matrix; the exact local coverage above is therefore the basis of this approval. I found no blocking issue.

AI disclosure: I used OpenAI Codex to inspect the exact revisions and policy, construct and run the layout/value/autograd matrix, execute the focused tests, and draft this review. I verified the results and conclusion.
